code/game/objects/items.dm

```diff
@@ -374,6 +374,7 @@
 
 ///Anything unique the item can do, like pumping a shotgun, spin or whatever.
 /obj/item/proc/unique_action(mob/user)
+	SEND_SIGNAL(src, COMSIG_ITEM_UNIQUE_ACTION, user)
```

Comment:
in general, if you are sending a signal at a parent proc, every child proc must call the parent proc. This signal was removed because it's impossible with the current gun architecture to call the unique action parent. 

Having this signal here would only cause confusion; a future coder might see this and use it, and not discover that guns actually don't use it.

An alternative i can suggest is using right click for the unique action stuff on vali, it might be way more intuitive that way; and you wouldn't have this issue with the signal